server/server.go

```diff
@@ -44,6 +44,7 @@ var Set = wire.NewSet(
 // The zero value is a server with the default options.
 type Server struct {
 	reqlog        requestlog.Logger
+	Handler       http.Handler
```

Comment:
Why is this field exported and others aren't?

Reply:
the intent was to allow the handler to be assigned after construction, since the previous flow was to pass it in to `ListenAndServe`.

Reply:
@eliben it's not clear to me if this was a imperative to change this to be unexported or if the question implies the existence of potentially valid reason(s) to export it, can you clarify?

Reply:
I was just wondering because it's different from the others. Your comment makes sense to me, but on the other hand it's hard to imagine why handlers would be changed for a server after creation. This is consistent with `http.Server` which is good, but then `http.Server` has all its fields public and here we hide all but `Handler`.
In any case I don't feel strongly about this and will defer to you and @zombiezen for a decision

Reply:
I was curious about this field being exported because it does introduce potential for it to be mutated, which I don't see a use case for yet. (I tend to lean on the side of keeping as much unexported as possible until someone has a concrete need, and then explicitly design for that.)

Reply:
unexported it, PTAL